content/docs/develop/userscripts/about-userscripts.md

```diff
@@ -0,0 +1,136 @@
+
+Example manifest:
+```json
+{
+  "name": "Scratch Messaging",
+  "description": "Provides easy reading and replying to your Scratch messages.",
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["projects", "https://scratch.mit.edu/", "profiles"]
+    }
+  ],
+  "tags": ["community"],
+  "enabledByDefault": false
+}
+```
+
+## Creating your first userscript
+
+Unlike extension content scripts and Tampermonkey userscripts, you must wrap all of your code inside a module default export:
+```js
+// Example userscript
+export default async function ({ addon, console }) {
+  console.log("Hello, " + await addon.auth.fetchUsername());
+  console.log("How are you today?");
+}
+```
+
+Remember that JavaScript allows functions to be declared inside other functions, for example:
+```js
+export default async function ({ addon, console }) {
+  async function sayHelloToUser() {
+    console.log("Hello, " + await addon.auth.fetchUsername());
+  }
+
+  await sayHelloToUser();
+  console.log("How are you today?");
+}
+```
+
+{{< admonition info >}}
+You can access many `addon.*` API utilities from userscripts. For example, you can get the current username, wait until an element exists on the page, or get a reference to the Scratch VM object.
+
+For more information, check the [API reference](/docs/reference/addon-api/).
+{{< /admonition >}}
+
+
+## Modifying the document HTML
+
+Use [browser DOM APIs](https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API) to customize the HTML of the page.
+
+Here's an example:
+```js
+const myButton = document.createElement("button");
+myButton.textContent = "Click me!";
+myButton.classList.add("button");
+myButton.setAttribute("title", "You're hovering a button");
+
+const myContainer = document.querySelector(".container");
+myContainer.append(myButton);
+```
+
+## Localizing userscripts
+
+Addon userscripts sometimes need to reference English words or sentences. Make sure not to hardcode them, so that they can be part of the translation process.
+
+{{< admonition error >}}
+```js
+// Don't do this:
+document.querySelector(".sa-find-bar").placeholder = "Find blocks";
+```
+{{< /admonition >}}
+
+To create a translatable string, follow these steps:
+1. Create a file named `addon-id.json` inside the `/addon-l10n/en` folder.
+2. Provide an ID for every string:
+```json
+{
+  "addon-id/find": "Find blocks"
+}
+```
+3. Make sure to import the `msg()` function in your userscript. The first line of your userscript should look like this:
+```js
+export default async function ({ addon, console, msg  }) {
+                                              // ^^^
+```
+4. Use the `msg()` function in your code, instead of a hardcoded string:
+```js
+document.querySelector(".sa-find-bar").placeholder = msg("find");
+```
+
+{{< admonition info >}}
+For more information about localizing userscripts, see [this page](/docs/localization/localizing-addons/).
+{{</admonition >}}
```

Comment:
This is probably a non-issue, but GitHub is complaining with the code highlighter not working well.
```suggestion
{{< /admonition >}}
```